incubator/sparkoperator/templates/crd-cleanup-job.yaml

```diff
@@ -0,0 +1,44 @@
+{{ if .Values.installCrds }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "sparkoperator.fullname" . }}-crd-cleanup
+  annotations:
+    "helm.sh/hook": pre-delete, pre-upgrade
```

Comment:
We shouldn't delete the CRDs when upgrading the operator. Deletion of the CRDs will cause all resources of the CRD types to be deleted.

Reply:
Ok that makes sense. But what if during an upgrade, the CRD types have changed? Without deleting the old types, it there a way to replace the old ones with the new ones?

Reply:
The operator by default creates or updates the CRDs at startup.

Reply:
Yeah I saw that in the operator code. But somehow during an upgrade, the CRD types were not upgraded for us.

Reply:
Updated the PR to delete the CRD only when running `helm delete`.